processor/transformer/transformer_test.go

```diff
@@ -180,13 +166,43 @@ func TestTransformer(t *testing.T) {
 		}
 
 		for _, tt := range tc {
+			statsStore, err := memstats.New()
+			require.NoError(t, err)
+
+			tr := handle{}
+			tr.stat = statsStore
+			tr.logger = logger.NOP
+			tr.conf = config.Default
+			tr.client = srv.Client()
+			tr.guardConcurrency = make(chan struct{}, 200)
```

Comment:
[question] do we have tests with way less concurrency as well to see if something gets stuck?